From 0c4694ef0743cea44cb8bbdd68b085fe11cae1ac Mon Sep 17 00:00:00 2001
From: Gajanan Bhat <gajanan.bhat@intel.com>
Date: Fri, 9 Oct 2015 12:01:43 +0530
Subject: [PATCH] REVERTME [VPG]: drm/i915: Re-enable primary if flip done is
 set on primary plane

Flip done on sprite plane will disable primary plane. Disable sprite plane is
not the only means of enabling back primary, even if flip done is set back on
primary plane, it should be enabled.

REVERTME: CHT product kernel relies on internal implementation of
atomic flip. Atomic flip is already addressed in a clean fashion using
nuclear flip in upstream and is used for Broxton and beyond.
So this patch will be a REVERTME patch and can be discarded in gmin_dev &
broxton trees and future forklifts.

Change-Id: I8057a1dc35b043ae6550b9ba8e6b443b92c4ce92
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-5799
Signed-off-by: Arun R Murthy <arun.r.murthy@intel.com>
Signed-off-by: Gajanan Bhat <gajanan.bhat@intel.com>
---
 drivers/gpu/drm/i915/intel_display.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index ab67a78..9adec40 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -10788,6 +10788,14 @@ static int intel_crtc_page_flip(struct drm_crtc *crtc,
 	if (work == NULL)
 		return -ENOMEM;
 
+	/* If page flip event flag is set, primary should be enabled */
+	if (page_flip_flags & DRM_MODE_PAGE_FLIP_EVENT) {
+		intel_crtc->pri_update = true;
+		intel_crtc->reg.cntr = DISPLAY_PLANE_ENABLE;
+		dev_priv->pipe_plane_stat |=
+			VLV_UPDATEPLANE_STAT_PRIM_PER_PIPE(intel_crtc->pipe);
+	}
+
 	work->event = event;
 	work->crtc = crtc;
 	work->old_fb = old_fb;
-- 
1.9.1

